src/core/lintString.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+/**
+ * @description wrapper for parse for programmatic use
+ * @param {String} string the text to parse
+ * @param {Object} [config] an object containing config to lint by
+ * @param {String} [filename] the name of the file
+ * @returns {Object} an object containing the complete report from linting
+ */
+var lintString = function( string, config, filename ) {
```

Comment:
Should have answered inline, cut&paste:

Yeah, but you might want a filename in the report.

Say the gulp plugin, it has a file, but Stylint won't have to do IO, as it's already read. But it still corresponds to a file.

Eslint also takes a (optional) filename: http://eslint.org/docs/developer-guide/nodejs-api#executeontext